### DISCUSSION TITLE
Problems deploying Ubuntu from the created template

### DISCUSSION THREAD

**Discussion by fperillo** [vmware/packer-examples-for-vsphere#837] [Help]
I created the template for Ubuntu 22.04 lts on VMware 7.0

I cloned the created template to a VM applying the configuration template that asks me IP, net mask, etc...

The VM starts with the correct IPs but at next reboot, cloud-init runs again and reverts to DHCP. I may disable cloud-init but how ?

A collegue tried to deploy with terraform and he had the same problem. SLES doesn't have the problem, but it doesn't use cloud-init... :-)

Which is the correct way to deploy a template in ubuntu with static IP assigned during deploy ?

**Comment by tenthirtyam** [comment]
For Ubuntu, it's enabled for cloud-init only and not for customization with guest tools. You'd need to modify the Ansible playbook to disable.

**Reply by tenthirtyam** [reply]
https://github.com/vmware-samples/packer-examples-for-vsphere/issues/269

**Reply by tenthirtyam** [reply]
And here's [a reference for an older](https://vmware-samples.github.io/packer-examples-for-vsphere/release-notes/?h=cloud+init#v2111) version that you might be able to adapt for ubuntu.

The other option is to have apt uninstall cloud-init at the end of the build process instead of configuring it for use.

**Reply by fperillo** [reply]
Thank you, I will check the old version tomorrow.

As last test today I modified the playbook creating /etc/cloud/cloud-init.disabled but the VM did not get the IP. 

Probably I need to disable the - name: "Configuring cloud-init." section of the ansible playbook.

**Reply by fperillo** [reply]
I understand that cloud-init can be really useful in a cloud environment but we just have VMware , at least up to the end of 2025 when current maintenance will expire... ;-)

I read several posts and since I don't want to change one line of this project, I added a new task to my "local" role (see my old post) with a apt: name: cloud-init state: absent


So cloud-init is uninstalled just before shutdown, IP is set via VMware, and we are happy :-)

I don't know why ubuntu is set to use cloud-init, probably we can add a switch to enable or disable it....

**Reply by tenthirtyam** [reply]
I've got an idea for this one that I'll add and test shortly. It's been on my to do list for a while. I maintain this in some free time. :)
